src/Gui/CommandStd.cpp

```diff
@@ -383,6 +385,7 @@ void StdCmdDlgPreferences::activated(int iMsg)
     if (cDlg.exec()) {
         cDlg.activeGroupPage(groupName, index);
     }
+    cDlg.isRebootRequired(); //The user may have applied first, then clicked the cancel button so it's not in the if(cDlg.exec())

```

Comment:
This is bad design. If the dialog recommends a reboot or not is an implementation detail and should not be exposed as public function.

The correct way is to handle this inside DlgPreferencesImp::accept() and DlgPreferencesImp::reject()

Reply:
Got it.